### ISSUE TITLE
Analog clock variant/addition?

### ISSUE THREAD

**Issue by exec228** [White-Tiger/T-Clock#143]
Salut!
Is there any hope for analog clock addition?
You know, clock hands are more intuitive for time periods estimation.
Thank you!
(I had some freeware item once, in 98/2003 epoch. It really draw vector hands in notification area. But I lost that freeware (sigh... it came with the source). And for several last years I cannot cannot google back that program or any similar one that supports Windows classic no-skin theme - only your product is in first 500 Google results :-))

**Comment by blackcrack** [comment]
did you ever use the T-Clock ? ??? This is not on Desktop, this is in the Systray,
a small 8 mm high thin clock in the taskbar .. holy ! (oh man.. common, can't be real now)

Can anyone close this tread and kick it in the file paper basket ?
best
Blacky

**Comment by exec228** [comment]
Yes! Definitely! I used TClock and fork TClockLite(Kazuto Sato) during an amount of two years or more, since 2005 or 2010.

Sure! Absolutely! I really used that another clock proggie with ANALOG hands, and YES! in system notification area (which you erroneously call SysTray)!

100%! Of-course! Both hour and minute hands were clearly visible in 32 pixels high image! Even with hour ticks around!
And I MEAN that it was Win95/98 theme, not XP/Vista theme.

Nope! Alas! If I could find it again via it's name, I would not ask here. I just have been using run THAT software without posting this question already!

**Comment by blackcrack** [comment]
\*vbg\* oohkey, calm down \*bg\*
we are both oldys ! \*vbg\* this i have see now :+1:  ;) 

yes, the name would be very helpful for search :)
and if i have it, could i find it maybe on any old ftp or so..
you know.. "the world of the 'soft'things" ;) 

(small tip, the jocker\* can you display with \\* )

best regards
blacky

blackysgate.de

**Comment by exec228** [comment]
Oh, another rare sample.
https://www.nakka.com/soft/trayc/
If I remember correctly, this project had too "bold" dial pad, over-designed, and not customized. I have chosen for usage another one which I meant in topic-start.

That another had the similar "high frequency name" which is extremely hard to google.

**Comment by White-Tiger** [comment]
what exactly do you expect T-Clock to do?
Adding some Notification Icon as an additional analog clock, or playing it where the digital clock is?

If I understood you correctly, your lost friend used the notification area / icons

**Comment by exec228** [comment]
Good day!

Thank you for response!

I suppose that option {No analog clock, Analog dial pad left of text of digital, Analog dial pad right of text of digital, Analog dial pad only} could be interesting. If it makes sense.

Personally I never created animated tray icons myself, and I cannot suggest whether graphical icon is currently "compatible" with text clock.

Another version is displaying multiple clocks: analog for UTC, digital for Paris, analog for local, digital (with date) for local, N clocks in total.

Yet, it's my personal opinion only :-)

P.S. Yes, I know that order of icons in real notification area is not determined, and user or system could and would rearrange icons randomly and unpredictably. Therefore, idea of multiple timezones is not necessarily a good idea.

**Comment by blackcrack** [comment]
hmm.. if you add more space by side of the text clock frame (so inside of it) and make maby more space right or left sideed, could you maybe handle that .... and therewith coult you make inside of the t-clock-frame an other frame with the clocks , left or right sided... or up /down,if the panel left or right sided, therewith could you determine  {up|left|bottom|right of the T-clock with an extra frame in your t-clockframe, therewith have you the possible for rearrange .. if you work with a headframe and frames inside of the frame.. like in html/php but (c or c++ of which programmer-lang is T-Clock scripted ? )

Animated Gif or png is maby not so an good idea, maybe more working with png's and then with y/x,
therewith can you give the possible for different skins (underfolder, named.png's) 
clockface (maybe like this https://upload.wikimedia.org/wikipedia/commons/thumb/3/3a/Ole_Mathiesen_OM1.jpg/521px-Ole_Mathiesen_OM1.jpg ), howerhand.png, minuethand. and secoundhand.png and the folder could be the themename(very easy holded ), therewith it is for the users possible to create 1000's of different Clocks for it.. also the swish Trainstop clock (some peoples be fan of it *giggle*)
best regards
Blacky

**Comment by exec228** [comment]
2. As for Nakka's clock, it's written in good'ol-C.

The programmer can set notification icon in run-time, taking it from resources file/segment. He "can do it all day".
On the other hand, icon is a bitmap which could be manipulated pixel-by-pixel.

1. That, let's call it "lost-tray-analog-clock", code was in C too.
When I looked into it's source, I recognised classical Bresenham's implementation of "line" function. That source was even shorter than Nakka's clock, containing one .c-file, one .ico-file just for shortcut demonstration purposes, and one project file for compiler. Creating actual icon for notification area "on the fly".
Therefore lost-tray-analog-clock hasn't a background picture prepared beforehand.

0. TClock is written on plain C as well.

**Comment by White-Tiger** [comment]
> TClock is written in plain C as well.

well the AutoVersion helper is partially C++ and the drawing rewrite is likely to make use of C++ as well.. _(because of GDI+)_

**Comment by blackcrack** [comment]
more time, more news, more information's and therewith more sugar and therewith.. more Fad !
holy you're right.. if i look in the Mirror.. damn !! \*lol\*
hihi.. humm.. well.. QT.. c++ and thereewith usable in Linux/plasma too..
so, that's the spirit or what is the stinginess ?, either one goes with the time or any when functioning  simply not anymore .. 

best regards
Blacky

**Comment by sqerk** [comment]
A slightly related request, I'd like to see a static icon.
I'm using T-Clock cause I can't stand staring at the clock all day, but I want to have the time and date a click away. Couldn't find a good way to do it in Windows 10, its just off or on.
So I've got T-Clock just showing me AM or PM with a fancy font, functions as my icon, is perfect.
Another idea is maybe icons that change a couple times a day: Morning, Day, Evening, Night
Just 4 icons

**Comment by White-Tiger** [comment]
well it's not really related to this issue... and I wouldn't directly call it "icons" either, but your request pretty much matches an idea I had for the new formatting system _(if it'll ever happen)_ as that'll allow to have different presets for how the clock looks.. eg. different, more detailed on mouse hover.
And since they're presets, I don't see a reason why mouse clicks, alarms, timers etc. shouldn't be able to trigger them as well. _(would require further rewrite of the alarm system as well.. though that's also something that should be done at some point)_
